### DISCUSSION TITLE
Is the data loader's per-key context allowed to be null?

### DISCUSSION THREAD

**Discussion by skingsland** [graphql-java/java-dataloader#94] [Q&A] [answered]
Based on looking at the code and testing it, it _appears_ that this data fetcher will pass through the key context to the data loader, _even if it's null_:

```
    TypeRuntimeWiring.newTypeWiring("Query")
        .dataFetcher(
            "myField",
            environment -> {
              String myId = environment.getArgument("myId");
              
              // it's OK if this is null, because the keyContext is allowed to be null?
              Integer myKeyContext = getNullableKeyContext(...);

              return environment.getDataLoader("myLoader").load(myId, myKeyContext);
            })
        .build();
```

Can you confirm, that though? Because nothing I found in the Javadocs or https://www.graphql-java.com/documentation/v16/batching#passing-context-to-your-data-loader says whether the per-key context is allowed to be `null`.

**Comment by bbakerman** [comment, marked as answer]
Yes they can be null
